connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/IncrementalCooperativeAssignor.java

```diff
@@ -245,10 +245,9 @@ ClusterAssignment performTaskAssignment(
         ConnectorsAndTasks deleted = diff(previousAssignment, configured);
         log.debug("Deleted assignments: {}", deleted);
 
-        // Derived set: The set of remaining active connectors-and-tasks is a derived set from the
-        // set difference of active - deleted
-        ConnectorsAndTasks remainingActive = diff(activeAssignments, deleted);
-        log.debug("Remaining (excluding deleted) active assignments: {}", remainingActive);
+        // The connectors and tasks that are currently running on more than one worker each
```

Comment:
Chris, this is something I get confused a lot with and not a comment for this PR but in general. There are a lot of steps which happen before we get to actual revocations. Can we just add a comment to explain how we get to it? Like `configured - duplicate - deleted` . I know it's all in the code but becomes difficult to follow at times ( could be due to my limited working with this class as well). Let me know what you think about it.

Reply:
I've found this part of the code base difficult to read and reason about as well. I'm not sure what would be helpful with this comment in particular, though?

If there's something in this PR that would benefit from extra comments, I'm happy to add them here until things look good. For other parts, we should probably leave that for a cosmetic follow-up that reduces the diff and allows us to focus on readability (and possibly make other, more aggressive, refactoring).

Reply:
What I meant was how do we start from the assignments and reduce it to the set with which we finally operate. I look at the code again and it's definitely less verbose in this PR v/s what it is in trunk. I can't call out what exactly to comment, but something that would help people figure out faster of the steps performed. We can take it later as well, not a problem.

Reply:
👍 sounds good. A few things we might consider:
- Compute derived sets immediately before they're used, instead of in a single large block at the beginning of the method
- Reduce comment size by removing the copy+pasted "Derived set: The set of... is a derived set from the set..." snippet
- Remove the `WorkerLoad` class and try to consolidate its logic into the `ConnectorsAndTasks` class (and the various `Map<String, ConnectorsAndTasks>` instances that we use)

(Disclaimer: haven't put much thought into this, these could be valuable, or we could find out in practice that they either aren't that useful or even actively make things worse)

Reply:
Yeah. I think these are not very important right now and we can have follow up JIRA tickets on this if needed to clean some things up.